<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NetworkPref</class>
 <widget class="QWidget" name="NetworkPref">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>501</width>
    <height>495</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Ports &amp;&amp; Limits</string>
     </property>
     <layout class="QGridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Port:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="kcfg_port">
        <property name="toolTip">
         <string>&lt;p&gt;Port used for the bittorrent protocol.&lt;/p&gt;
&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Attention:&lt;/span&gt; If you are behind a router, this port needs to be forwarded to accept incoming connections. The UPnP plugin can do this for you.&lt;/p&gt;</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>3000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>UDP tracker port:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="kcfg_udpTrackerPort">
        <property name="toolTip">
         <string>&lt;p&gt;Port used for the UDP tracker protocol.&lt;/p&gt;
&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Attention:&lt;/span&gt; If you are behind a router, this port needs to be forwarded. The UPnP plugin can do this for you.&lt;/p&gt;</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>65536</number>
        </property>
        <property name="value">
         <number>4000</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Maximum connections per torrent:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="kcfg_maxConnections">
        <property name="toolTip">
         <string>The maximum number of connections allowed per torrent.</string>
        </property>
        <property name="specialValueText">
         <string>No limit</string>
        </property>
        <property name="maximum">
         <number>99999</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Global connection limit:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="kcfg_maxTotalConnections">
        <property name="toolTip">
         <string>The global connection limit for all torrents combined.</string>
        </property>
        <property name="specialValueText">
         <string>No limit</string>
        </property>
        <property name="maximum">
         <number>99999</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Maximum download speed:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QSpinBox" name="kcfg_maxDownloadRate">
        <property name="toolTip">
         <string>The maximum download speed in KiB/s.</string>
        </property>
        <property name="specialValueText">
         <string>No limit</string>
        </property>
        <property name="suffix">
         <string> KiB/s</string>
        </property>
        <property name="maximum">
         <number>6500000</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Maximum upload speed:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QSpinBox" name="kcfg_maxUploadRate">
        <property name="toolTip">
         <string>The maximum upload speed in KiB/s</string>
        </property>
        <property name="specialValueText">
         <string>No limit</string>
        </property>
        <property name="suffix">
         <string> KiB/s</string>
        </property>
        <property name="maximum">
         <number>6500000</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Advanced</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QGridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>DSCP value for IP packets:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="kcfg_dscp">
          <property name="toolTip">
           <string>This value will be filled in the DSCP field of all the IP packets sent for the bittorrent protocol.</string>
          </property>
          <property name="maximum">
           <number>63</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Maximum number of connection setups:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="kcfg_maxConnectingSockets">
          <property name="toolTip">
           <string>&lt;p&gt;The maximum number of outgoing connections ktorrent will try to setup simultaneously.&lt;/p&gt;
&lt;p&gt;If you are having trouble with ktorrent blocking other internet traffic, try setting this number a bit lower.&lt;/p&gt;</string>
          </property>
          <property name="minimum">
           <number>10</number>
          </property>
          <property name="maximum">
           <number>200</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Network interface:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="combo_networkInterface">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>&lt;p&gt;Which network interface to listen on for incoming connections.&lt;/p&gt;
&lt;p&gt;&lt;span style=&quot;font-weight:600;&quot;&gt;Note: &lt;/span&gt;Requires a restart to take effect.&lt;/p&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="kcfg_utpEnabled">
        <property name="toolTip">
         <string>The µTorrent transport protocol runs over UDP and is optimized to not interfere with other traffic, while still using up all unused bandwidth.</string>
        </property>
        <property name="text">
         <string>Use the µTorrent transport protocol (µTP)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="kcfg_onlyUseUtp">
        <property name="toolTip">
         <string>Don't use TCP for bittorrent connections, only use the µTP protocol. This can lead to slower speeds, seeing that not all bittorrent clients support µTP.</string>
        </property>
        <property name="text">
         <string>Only use µTP</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Primary transport protocol:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="kcfg_primaryTransportProtocol">
          <property name="toolTip">
           <string>The primary transport protocol to use, when attempting to setup a connection with a peer, this protocol is chosen first.</string>
          </property>
          <item>
           <property name="text">
            <string>TCP</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>µTP</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="m_recommended_settings">
       <property name="toolTip">
        <string>Let KTorrent calculate the recommended settings based upon the bandwidth you have available.</string>
       </property>
       <property name="text">
        <string>Recommended Settings...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>493</width>
       <height>41</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
